### PR TITLE
Add direct grants lite

### PR DESCRIPTION
**Pull request by 0xKurt**
Locally all the changes have been tested on v1 and v2.
Currently, the contract is deployed only on sepolia.

Post fixing the audit changes, we will deploy to all chains and have another PR which disables direct grants round creation on v1 and enables only on v2

We also removed the legacy subgraph calls

Flows Tested:
http://localhost:3000/#/round/116

- :white_check_mark: Created DG on v2
- :white_check_mark: Edit DG on v2
- :white_check_mark: Apply on DG v2
- :white_check_mark: Approve for review on v2
- :white_check_mark: Reject on V2
- :white_check_mark: Approve on V2
- :white_check_mark: Payout on V2
- :white_check_mark:  View stat for payout for v2

image

- :white_check_mark: Created DG on v1
- :white_check_mark: Edit Metadata DG on v1
- [x] Edit Timestamps DG on v1
- [x] Apply on DG v1
- [x] Approve for review on v1
- [x] Reject on V1
- [x] Approve on V1
- [x] Payout on V1
- [x] View stat for payout for v1

### CODE REVIEW

File: packages/common/src/allo/backends/allo-v1.ts [hunk 1]
```diff
@@ -1173,6 +1173,31 @@ export class AlloV1 implements Allo {
...
+  }): AlloOperation<
+    Result<{ blockNumber: BigInt }>,
+    {
+      transaction: Result<Hex>;
+      transactionStatus: Result<TransactionReceipt>;
+      indexingStatus: Result<void>;
+    }
+  > {
+    return new AlloOperation(async ({ emit }) => {
+      console.log("V1 is not supported yet");
```
Comment (thelostone-mc) [inline packages/common/src/allo/backends/allo-v1.ts:1194]: ^ This doesn't seem to execute

File: packages/round-manager/src/context/application/usePayout.ts [hunk 2]
```diff
@@ -124,34 +130,77 @@ export function usePayout() {
...
+    //   roundId,
+    //   signer
+    // );
+    // let txBlockNumber: number;
     try {
-      setContractUpdatingStatus(ProgressStatus.IN_PROGRESS);
-      const payoutReceipt = await directPayout.payout({
-        vault: payoutVault,
+      console.log("Payout 2");
+      await allo.payoutDirectGrants({
```
Comment (thelostone-mc) [inline packages/round-manager/src/context/application/usePayout.ts:143]: and the editor points out that await is not required here (which is weird cause we do an await on all other functions in allo.ts)

File: packages/common/src/allo/backends/allo-v2.ts [hunk 3]
```diff
@@ -1196,6 +1207,86 @@ export class AlloV2 implements Allo {
...
+    applicationIndex?: number;
+  }): AlloOperation<
+    Result<{ blockNumber: BigInt }>,
+    {
+      transaction: Result<Hex>;
+      transactionStatus: Result<TransactionReceipt>;
+      indexingStatus: Result<void>;
+    }
+  > {
+    console.log("pay 0-0");
```
Comment (thelostone-mc) [inline packages/common/src/allo/backends/allo-v2.ts:1227]: this prints

File: packages/common/src/allo/backends/allo-v2.ts [hunk 4]
```diff
@@ -1196,6 +1207,86 @@ export class AlloV2 implements Allo {
...
+    return new AlloOperation(async ({ emit }) => {
+      console.log("pay 0");
```
Comment (thelostone-mc) [inline packages/common/src/allo/backends/allo-v2.ts:1229]: ^ This doesn't seem to execute @boudra

File: packages/round-manager/src/context/application/usePayout.ts (hunk 2 above)
Comment (boudra) [reply packages/round-manager/src/context/application/usePayout.ts:143]: you need to execute the operation by running `execute`, check other calls to Allo.

File: packages/round-manager/src/features/api/utils.ts [hunk 5]
```diff
@@ -341,3 +341,6 @@ export const formatCurrency = (
     maximumFractionDigits: fraction || 3,
   });
 };
+
+export const isInfiniteDate = (roundTime: Date) =>
+  roundTime.getFullYear() === 2243;
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/api/utils.ts:346]:
```suggestion
  const UINT_64_MAX_YEAR = 2243; 
  roundTime.getFullYear() === UINT_64_MAX_YEAR;
```

File: packages/round-manager/src/features/program/ViewProgramPage.tsx [hunk 6]
```diff
@@ -256,82 +256,83 @@ export default function ViewProgram() {
...
-                  <div
-                    className={`rounded-full border ${
-                      grantType === "directGrant"
-                        ? "border-violet-400"
-                        : "border-grey-100"
-                    } h-[24px] w-[24px]`}
-                    style={{
-                      borderWidth: grantType === "directGrant" ? "6px" : "2px",
-                    }}
+            {getAlloVersion() === "allo-v1" && ( // todo: set to v2 again
```
Comment (thelostone-mc) [inline packages/round-manager/src/features/program/ViewProgramPage.tsx:259]: FIX

File: packages/common/src/allo/backends/allo-v2.ts [hunk 7]
```diff
@@ -53,15 +59,15 @@ function getStrategyAddress(strategy: RoundCategory, chainId: ChainId): string {
       strategyAddresses = {
         [RoundCategory.QuadraticFunding]:
           "0x029dFAf686DfA0efdace5132ba422e9279D50b5b",
-        [RoundCategory.Direct]: "0x45181C4fD52d4d350380B3D42091b80065c702Ef",
+        [RoundCategory.Direct]: "0x0000000000000000000000000000000000000000",
```
Comment (thelostone-mc) [inline packages/common/src/allo/backends/allo-v2.ts:62]: This has been set cause we need to redeploy the new strategy post audit fixes

**Review by thelostone-mc — APPROVED** [review]

**Review by codenamejason — APPROVED** [review]

**Comment by meglister** [comment]
Results of my testing for V2 on Sepolia

[x] successfully created a direct grants round
[ ] round displays with incorrect info -- labeled as Quadratic Funding and does not reflect accurate start date (I think it should be 5/10/24)
Screenshot 2024-05-08 at 9 36 12 PM
[ ] was not able to update direct grants round, console error attached
Screenshot 2024-05-08 at 9 38 27 PM
[ ] round does not display correctly in explorer. should not show donate dates or matching pool amounts. https://explorer-staging-git-add-direct-grants-lite-grants-stack.vercel.app/#/round/11155111/133
Screenshot 2024-05-08 at 9 39 47 PM
[ ] unable to apply to round because of invalid date issue https://builder-staging-git-add-direct-grants-lite-grants-stack.vercel.app/#/chains/11155111/rounds/133
Screenshot 2024-05-08 at 9 41 53 PM

**Comment by meglister** [comment]
I did validate that QF is working as expected in the branch with DG changes (on v2)
- [x] created a round
- [x] applied to a round
- [x] edited a round
- [x] accepted application
- [x] rejected application

**Comment by meglister** [comment]
Testing DG in Allo v1
- [ ] unable to deploy round with open end date 
Screenshot 2024-05-08 at 10 07 09 PM
- [ ] unable to deploy round (end date is set) but not getting any errors, it just hangs? It's been over 5 min. Here's a link to my wallet in case it does wind up going through. https://sepolia.etherscan.io/address/0x58338E95caEf17861916Ef10daD5fAFE20421005

**Comment by codenamejason** [comment]
> DG v1
> 
> * [x]  created a round
> * [x]  edited a round
> * [x]  applied for a round
> * [x]  approved applications
> * [ ]  cannot pay out applications -- see screenshot and error attached. I have a blank screen that loads
>   Screenshot 2024-05-09 at 2 01 36 PM

good catch! we need to prevent the user from removing required questions.

**Review by melissa-neira — CHANGES_REQUESTED** [review]
small thing:

Screenshot 2024-05-09 at 20 26 39

- the round status badges are behaving strangely 

<br>
I was able to successfully deploy a round on sepolia, so yay!

**Comment by 0xKurt** [comment]
@melissa-neira 
the badges are already off since a while. maybe you want to create a ticket for it with a link/image of how they should look like?

the new strategy is currently not deployed to arbitrum, only sepolia

**Comment by 0xKurt** [comment]
> DG v1
> 
> * [x]  created a round
> * [x]  edited a round
> * [x]  applied for a round
> * [x]  approved applications
> * [ ]  cannot pay out applications -- see screenshot and error attached. I have a blank screen that loads
>   Screenshot 2024-05-09 at 2 01 36 PM

i faced this one today: https://github.com/gitcoinco/grants-stack/issues/3424
which is kinda related to your problem with the removed token.

i have an idea on how to solve it and add the new feature with the token list. maybe we can talk about it next week really quick

**Comment by meglister** [comment]
DG v2
- [x] created round
- [x] edited round
- [x] applied to round
- [ ] I marked an application as in review from the application page in manager. The tx succeeded, but it was not reflected in the UI. Here's the tx: https://sepolia.etherscan.io/tx/0x66509ef35737d7d5973f81750f490347c82c08d33056cce741253524dda6720a
Screenshot of UI behind processing modal -- I tried again to approve from the Grant Applications page and after the page refresh the new status was reflected. Maybe we need to force a page refresh or something.
Screenshot 2024-05-09 at 2 51 32 PM
- [x] approved applications
- [ ] paid out applications -- not working because of same error as above with removing payout token field
Screenshot 2024-05-09 at 2 53 58 PM
- [ ] view applications in explorer -- open end dates are not working as expected
Screenshot 2024-05-09 at 2 48 55 PM

**Comment by codenamejason** [comment]
> small thing:
> 
> Screenshot 2024-05-09 at 20 26 39
> * the round status badges are behaving strangely
> 
> 
> I was able to successfully deploy a round on sepolia, so yay!

The was done to accomidate the 'applications not started' badge, we can revisit how to render the badges in a nice consistent way.

**Comment by 0xKurt** [comment]
@meglister 
The issue with the explorer timestamps appears because if you click the link you open the explorer in production and not staging.

Onq staging it's already fixed https://explorer-staging-git-add-direct-grants-lite-grants-stack.vercel.app/#/round/11155111/139

**Pull request merged**